static/js/apps/explore/page_overview.tsx

```diff
@@ -0,0 +1,174 @@
+ */
+
+/**
+ * Component for rendering the generated page overview.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { ReactElement, useEffect, useState } from "react";
+
+import { Loading } from "../../components/elements/loading";
+import { SubjectPageMetadata } from "../../types/subject_page_types";
+
+const GLOBAL_CAPTURE_LINK_GROUP = /<([^<>]+)>/g;
+const CAPTURE_LINK_GROUP = /<([^<>]+)>/;
+const SPLIT_LINKS = /(<[^<>]+>)/;
+const CHECK_MARKERS_EXIST = /<|>/;
+const GLOBAL_MARKERS = /<|>/g;
+
+type ChartIndex = {
+  title: string;
+  block: number;
+  category: number;
+};
+
+interface PageOverviewPropType {
+  query: string;
+  pageMetadata: SubjectPageMetadata;
+}
+
+export function PageOverview(props: PageOverviewPropType): ReactElement {
+  const [pageOverview, setPageOverview] = useState([]);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    const statVars = getRelevantStatVars(props.pageMetadata);
+    getPageOverview(props.query, statVars)
+      .then((value) => {
+        setPageOverview(value);
+      })
+      .catch(() => {
+        setPageOverview([]);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [props.query, props.pageMetadata]);
+
+  return (
+    <>
+      {loading && (
+        <div className="page-overview-loading">
+          <Loading />
+        </div>
+      )}
+      {pageOverview && (
+        <div className="page-overview-inner">{pageOverview}</div>
+      )}
+    </>
+  );
+}
+
+const getPageOverview = async (
+  query: string,
+  statVars: Array<ChartIndex>
+): Promise<Array<React.ReactNode>> => {
+  if (_.isEmpty(query) || _.isEmpty(statVars)) {
+    return [];
+  }
+  const url = "/api/explore/page-overview";
+  const body = {
+    q: query,
+    statVars: statVars.map((stat) => {
+      return stat.title;
+    }),
+  };
+  return await axios.post(url, body).then((resp) => {
+    // Check link syntax in overview
+    const preprocessedOverview = resp.data.page_overview;
+    const testStatSyntax = preprocessedOverview.replace(
+      GLOBAL_CAPTURE_LINK_GROUP,
+      "$1"
+    );
+    // If the markers still exist, the links were not marked properly so the overview with no links is returned
+    if (CHECK_MARKERS_EXIST.test(testStatSyntax)) {
+      const cleanedOverview = testStatSyntax.replace(GLOBAL_MARKERS, "");
+      return <span>{cleanedOverview}</span>;
+    }
+    // Create Maps to speed up string look ups
+    const preprocessedStatVarIndices = resp.data.stat_var_links;
+    const substringsToStatVarIndex = new Map<string, string>();
+    for (const statVarIdx of preprocessedStatVarIndices) {
+      substringsToStatVarIndex.set(
+        statVarIdx.natural_language,
+        statVarIdx.stat_var_title
+      );
+    }
+    const titleToCatNBlockIndex = new Map<string, ChartIndex>();
+    for (const statVarChart of statVars) {
+      titleToCatNBlockIndex.set(statVarChart.title, statVarChart);
+    }
```

Comment:
consider renaming; another option might be statVarTitleToChartConfig or statVarTitleToPageLocation
```suggestion
    const statVarTitleToChartLocation = new Map(
  statVarChartLocations(statVarChart => [
    statVarChart.title,
   statVarChart
  ])
);
```

Reply:
Done. I added it through a map function which I believe is what you're referring to.